src/http/Stream.cc

```diff
@@ -428,14 +428,12 @@ Http::Stream::buildRangeHeader(HttpReply *rep)
         range_err = "no [parse-able] reply";
     else if ((rep->sline.status() != Http::scOkay) && (rep->sline.status() != Http::scPartialContent))
         range_err = "wrong status code";
-    else if (hdr->has(Http::HdrType::CONTENT_RANGE)) {
-        if (rep->sline.status() == Http::scPartialContent)
-            range_err = "existing Content-Range in the 206 response";
-        else {
-            assert(rep->sline.status() == Http::scOkay);
-            range_err = "unexpected Content-Range in the 200 response";
-        }
-    }
+    else if (rep->sline.status() == Http::scPartialContent)
+        range_err = "too complex response"; // probably contains what the client needs
+    else if (rep->sline.status() != Http::scOkay)
+        range_err = "wrong status code";
+    else if (hdr->has(Http::HdrType::CONTENT_RANGE))
```

Comment:
We should mention `content_range` here to address my first review concerns. How about something like the following?

```
else if (hdr->has(Http::HdrType::CONTENT_RANGE)) {
    Must(!rep->content_range); // this is a 200, not 206 response
    range_err = "meaningless response"; // the status code or the header is wrong
}
```

Reply:
Done with an adjustment, please see the updated PR.